phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -450,7 +463,7 @@ private void closeStatements() throws SQLException {
         // from modifying this list.
         this.statements = Lists.newArrayList();
         try {
-            mutationState.rollback(this);
+            mutationState.clear();
```

Comment:
After the tx timeout it ends up on the invalid list, I will add a test for this.